src/components/Header/Header.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { Todo } from '../../types/todo';
+import { USER_ID, addTodo } from '../../api/todos';
+import classNames from 'classnames';
+
+type Props = {
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  setUpdatingTodosIds: (value: Set<number>) => void;
+  setErrorMessage: (message: string) => void;
+  setTempTodo: (tempTodo: Todo | null) => void;
+  changeTodoStatus: (todo: Todo) => void;
+};
+
+export const Header: React.FC<Props> = ({
+  todos,
+  setTodos,
+  setUpdatingTodosIds,
+  setErrorMessage,
+  setTempTodo,
+  changeTodoStatus,
+}) => {
+  const [newTitle, setNewTitle] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const allTodosCompleted = todos.every(todo => todo.completed);
+
+  // focus on the input field when it's not submitting and todos length changes
+  useEffect(() => {
+    if (!isSubmitting && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isSubmitting, todos.length]);
+
+  // handle form submission to add a new todo
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const title = newTitle.trim();
+
+    if (title === '') {
```

Comment:
```suggestion
    if (!title) {
```
nit: you can check it in this way